baseservice/base_service.py

```diff
@@ -0,0 +1,122 @@
+import logging
+import signal
+import threading
+from abc import ABCMeta, abstractmethod
+from enum import Enum, unique
+from typing import Optional
+
+from baseservice.utils import Event
+
+
+@unique
+class ServiceState(Enum):
+    STARTING = 'STARTING'
+    STOPPING = 'STOPPING'
+    STOPPED = 'STOPPED'
+
+
+class BaseService(metaclass=ABCMeta):
+    """
+    this class is the base class for all services
+    """
+
+    def __init__(self, *, name: str = None, should_stop_on_signal=True):
+        """
+
+        :param name: the name of this service. if None, the name of the type will be used
+        :param should_stop_on_signal: if True, the service will try to register SIGTERM and SIGINT on stop method.
+        """
+        self._should_stop_on_signal: bool = should_stop_on_signal
+        self._cancellation_token: threading.Event = threading.Event()
```

Comment:
Token is not an indicative name considering this is just an event

Reply:
cancellation_token is a well known name for this design pattern...